### PR TITLE
Score assignment and keypoint culling

### PR DESCRIPTION
**Pull request by SachitM**
Assigning Score to Possibly Dynamic and Repeatable Features
Lowered Threshold of inliers
Removing Dynamic Features (Will be moving the feature removal to ORB Extract)

### CODE REVIEW

File: orb_slam_2_ros/orb_slam2/src/Frame.cc [hunk 1]
```diff
@@ -449,6 +465,50 @@ void Frame::UndistortKeyPoints() {
...
 }
 
+void Frame::ScoreKeyPoints(const cv::Mat &semanticmap, bool enable_removal)
+{
+    //Remove Dynamic KeyPoints
+    std::vector<cv::KeyPoint> mvKeysTemp;
+    cv::Mat mDescriptorsTemp;
+    int idx_key = 0;
+    int enable = 1;
+    for(int i=0; i<N; i++)
```
Comment (heethesh) [inline orb_slam_2_ros/orb_slam2/src/Frame.cc:491]: where is N declared?

Comment (SachitM) [reply orb_slam_2_ros/orb_slam2/src/Frame.cc:491]: Its a member variable, updated right before this function is called

Comment (SachitM) [reply orb_slam_2_ros/orb_slam2/src/Frame.cc:491]: It's part of the original codebase

File: orb_slam_2_ros/orb_slam2/src/Frame.cc [hunk 2]
```diff
@@ -449,6 +465,50 @@ void Frame::UndistortKeyPoints() {
...
+            // mDescriptorsTemp.push_back(mDescriptors.row(0));
+            continue;
+        }
+        mvKeysTemp.push_back(mvKeys[i]);
+        mvScoreDynamic.push_back(float(pixel->y)/255.0);
+        mvScoreRepeatable.push_back(float(pixel->x)/255.0);
+        mDescriptorsTemp.push_back(mDescriptors.row(i));
+    }
+    
+    mvKeys = mvKeysTemp;
```
Comment (heethesh) [inline orb_slam_2_ros/orb_slam2/src/Frame.cc]: Are we sure this would do a deep copy?

File: orb_slam_2_ros/orb_slam2/src/Frame.cc [hunk 3]
```diff
@@ -449,6 +465,50 @@ void Frame::UndistortKeyPoints() {
...
+    mDescriptors = mDescriptorsTemp;
+    cerr << "UpdatedPoints "<< mvKeysTemp.size()<<":" << N<<endl;
+    
+    N = mvKeysTemp.size();
+
```
Comment (heethesh) [inline orb_slam_2_ros/orb_slam2/src/Frame.cc]: These should be `mvKeys.size()` in both lines above

File: orb_slam_2_ros/orb_slam2/src/Frame.cc [hunk 4]
```diff
@@ -449,6 +465,50 @@ void Frame::UndistortKeyPoints() {
   }
 }
 
+void Frame::ScoreKeyPoints(const cv::Mat &semanticmap, bool enable_removal)
+{
+    //Remove Dynamic KeyPoints
+    std::vector<cv::KeyPoint> mvKeysTemp;
+    cv::Mat mDescriptorsTemp;
+    int idx_key = 0;
+    int enable = 1;
```
Comment (heethesh) [inline orb_slam_2_ros/orb_slam2/src/Frame.cc]: Remove unused variables

File: orb_slam_2_ros/orb_slam2/src/Frame.cc (hunk 2 above)
Comment (SachitM) [reply orb_slam_2_ros/orb_slam2/src/Frame.cc]: Yes

**Pull request merged**